### DISCUSSION TITLE
Way to typecheck objects post-transform?

### DISCUSSION THREAD

**Discussion by Brandon-Perry** [colinhacks/zod#1840] [Q&A] [answered]
I have an object where when I parse I want to take its date fields and convert them to ISO strings. 
`
const myObj = z.object({
   myDate: z.date().transform(date => date.toISOString)())
})
`
However, I would like to later confirm that an object is the post transformed version later on. However, this results in a failure
`
const myFutureObj = {
   myDate: 'some ISO string'
}
myObj.parse(myFutureObj) <== fails
`

Is there a way I can test an object to see if it's the post transformed version?

**Comment by JacobWeisenburger** [comment, marked as answer]
Does something like this work for you?

```ts
const dateToString = z.date().transform( date => date.toISOString() )

console.log( dateToString.safeParse( new Date() ).success ) // true
console.log( z.coerce.date().safeParse( '2023-01-10T00:00:00.000Z' ).success ) // true
```
